### PR TITLE
[7618] follow button redirect

### PR DESCRIPTION
**Pull request by goapunk**

### CODE REVIEW

File: adhocracy4/follows/static/follows/__tests__/FollowButton.jest.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import React from 'react'
+import { render, fireEvent, screen } from '@testing-library/react'
+import '@testing-library/jest-dom'
+import { FollowButton } from '../FollowButton'
+import jQuery from 'jquery'
+import api from '../../../../static/api'
+window.$ = window.jQuery = jQuery
```
Comment (philli-m) [inline adhocracy4/follows/static/follows/__tests__/FollowButton.jest.jsx:7]: why do we need jquery in react test?

Comment (goapunk) [reply adhocracy4/follows/static/follows/__tests__/FollowButton.jest.jsx:7]: ah, is left over from mocking the ajax call in the api, will remove

File: adhocracy4/follows/static/follows/__tests__/FollowButton.jest.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  followingButton = await screen.findByText('Following')
+  expect(followingButton).toBeTruthy()
+  followButton = screen.queryByText('Follow')
+  expect(followButton).toBeNull()
+  expect(api.follow.change).toHaveBeenCalledTimes(1)
+})
+
+test('Test FollowButton redirect', async () => {
+  delete window.location
+  window.location = {}
```
Comment (philli-m) [inline adhocracy4/follows/static/follows/__tests__/FollowButton.jest.jsx:51]: what are these 2 lines doing? maybe a comment would be good unless its very obvious and i'm just missing it?!

Comment (goapunk) [reply adhocracy4/follows/static/follows/__tests__/FollowButton.jest.jsx:51]: will add comment

File: changelog/7618.md [hunk 3]
```diff
@@ -1,3 +1,5 @@
 ### Changed
 
 - refactor follow to be functional and add aria described by for when no alert shown and use a4 prefix for classes so external style liberies can be used (story !7618/7701)
+- redirect to login page when follow button is pressed and user is not logged
+  ing
```
Comment (m4ra) [inline changelog/7618.md:5]: react and js is not my cup of tea, so cannot review properly, but here is a typo :)

File: adhocracy4/follows/static/follows/FollowButton.jsx [hunk 4]
```diff
@@ -2,11 +2,16 @@ import django from 'django'
...
 import Alert from '../../../static/Alert'
 
-const api = require('../../../static/api')
+import api from '../../../static/api'
+import config from '../../../static/config'
 
 const translated = {
-  followDescription: django.gettext('Click to be updated about this project via email.'),
-  followingDescription: django.gettext('Click to no longer be updated about this project via email.'),
+  followDescription: django.gettext(
```
Comment (goapunk) [inline adhocracy4/follows/static/follows/FollowButton.jsx:9]: was auto-formatted

File: adhocracy4/static/__mocks__/api.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+let following = null
+
+const api = {
```
Comment (goapunk) [inline adhocracy4/static/__mocks__/api.js:3]: not sure if this is the best way but I had to mock the api

**Review by philli-m — APPROVED** [review]
Coolio!

**Pull request merged**